Odd pkgconfig issue when enabling wayland, needs explicit handling for wayland-client

Getting occasional errors where wayland-client.h cannot be found when building intel-vaapi-driver.

Distro: OpenSuse Tumbleweed

Build and install libva locally with wayland support

Build and install intel-vaapi-driver with wayland support

Consistently get this error:

  CC       libi965_drv_video_la-i965_output_wayland.lo
In file included from i965_output_wayland.c:30:0:
/usr/include/va/va_backend_wayland.h:31:28: fatal error: wayland-client.h: No such file or directory
 #include <wayland-client.h>
                            
Of course I have all the deps installed.  It's like it is only looking in /usr/include rather than also looking in /usr/include/wayland

A simple work around is to add  -I/usr/include/wayland to driver_cflags...

But that should not be necessary because the pkgconfig is clearly present and adds /usr/include/wayland to the includdir search path which in turn is added to wayland-client cflags.  

I don't have this problem on Arch so is it something about OpenSuse.

--- a/configure.ac
+++ b/configure.ac
@@ -188,6 +188,8 @@ if test "$enable_wayland" = "yes"; then
 
     if test "$USE_WAYLAND" = "yes"; then
 
+        PKG_CHECK_MODULES([WAYLAND_CLIENT], [wayland-client],
+        [INCLUDES="$INCLUDES $WAYLAND_CLIENT_CFLAGS"], [:])
         WAYLAND_PREFIX=`$PKG_CONFIG --variable=prefix wayland-client`
         AC_PATH_PROG([WAYLAND_SCANNER], [wayland-scanner],,
                      [${WAYLAND_PREFIX}/bin$PATH_SEPARATOR$PATH])

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -184,7 +184,7 @@ if USE_WAYLAND
 source_c			+= i965_output_wayland.c
 source_h			+= i965_output_wayland.h
 source_h			+= $(protocol_source_h)
-driver_cflags			+= $(WAYLAND_CFLAGS)
+driver_cflags			+= $(WAYLAND_CFLAGS) $(WAYLAND_CLIENT_CFLAGS)
 endif
 
 # git version

